environments/hca-prod/terraform/buckets.tf

```diff
@@ -83,16 +82,28 @@ resource google_storage_bucket ucsc_bucket {
   location = "US"
 }
 
-resource google_storage_bucket_iam_member ucsc_bucket_iam {
+resource google_storage_bucket_iam_member ucsc_bucket_reader_iam {
   provider = google-beta.target
   bucket   = google_storage_bucket.ucsc_bucket.name
   # When the storage.admin role is applied to an individual bucket,
   # the control applies only to the specified bucket and objects within
   # the bucket: https://cloud.google.com/storage/docs/access-control/iam-roles
-  for_each = toset(["hannes@ucsc.edu", "dsotirho@ucsc.edu"])
+  for_each = toset(["platform-hca-dev-read@ucsc.edu", "platform-hca-prod-read@ucsc.edu"])
 
   role   = "roles/storage.admin"
-  member = "user:${each.value}"
+  member = "group:${each.value}"
+}
+
+resource google_storage_bucket_iam_member ucsc_bucket_writer_iam {
+  provider = google-beta.target
+  bucket   = google_storage_bucket.ucsc_bucket.name
+  # When the storage.admin role is applied to an individual bucket,
+  # the control applies only to the specified bucket and objects within
+  # the bucket: https://cloud.google.com/storage/docs/access-control/iam-roles
+  for_each = toset(["platform-hca-dev-write@ucsc.edu", "platform-hca-prod-write@ucsc.edu"])
```

Comment:
❓ Why is the `platform-hca-dev-write@ucsc.edu` user getting objectAdmin access in the hca-prod env?

Reply:
From Hannes:

> Ideally we'd like the two read groups to be given read access to all staging areas. I don't think we cleanly differentiate between dev and prod staging areas. That why both groups to all SAs for reading.
> Both write groups should be given write access to all UCSC-owned staging areas.
They only have one bucket and it is in our prod env, they made dev/prod paths under it. I told him to give us one email list and he gave 4 so here we are.

Reply:
in that case, could we pop a `condition {}` block into these resources to limit their access to those paths?

Reply:
@aherbst-broad ^